examples/with-ethers/src/eip712/erc3009_transfer.ts

```diff
@@ -0,0 +1,108 @@
+import { createNewWallet } from "../createNewWallet";
+
+async function main() {
+  if (!process.env.SIGN_WITH) {
+    // If you don't specify a `SIGN_WITH`, we'll create a new wallet for you via calling the Turnkey API.
+    await createNewWallet();
+    return;
+  }
+
+  const turnkeyClient = new TurnkeyClient(
+    {
+      baseUrl: process.env.BASE_URL!,
+    },
+    new ApiKeyStamper({
+      apiPublicKey: process.env.API_PUBLIC_KEY!,
+      apiPrivateKey: process.env.API_PRIVATE_KEY!,
+    }),
+  );
+
+  // Initialize a Turnkey Signer
+  const turnkeySigner = new TurnkeySigner({
+    client: turnkeyClient,
+    organizationId: process.env.ORGANIZATION_ID!,
+    signWith: process.env.SIGN_WITH!,
+  });
+
+  // Bring your own provider (such as Alchemy or Infura: https://docs.ethers.org/v6/api/providers/)
+  const network = "goerli";
+  const provider = new ethers.InfuraProvider(network);
+  const connectedSigner = turnkeySigner.connect(provider);
+  const address = await connectedSigner.getAddress();
+
+  print("Address:", address);
+
+  // 4. Sign an EIP-712 Payload for a Hyperliquid `ApproveAgent` operation
+  const approveAgentPayload = {
+    types: {
+      TransferWithAuthorization: [
+        { name: "from",          type: "address"  },
+        { name: "to",            type: "address"  },
+        { name: "value",         type: "uint256"  },
+        { name: "validAfter",    type: "uint256"  },
+        { name: "validBefore",   type: "uint256"  },
+        { name: "nonce",         type: "bytes32"  },
+      ],
+    },
+    domain: {
+      name: "USD Coin",               // ERC-20 token name
+      version: "2",                   // USDC v2 implements EIP-3009 & EIP-2612
+      chainId: 1,                     // Ethereum Mainnet
+      verifyingContract: "0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48"
+    },
+    primaryType: "TransferWithAuthorization",
+    message: {
+      from:        "0x1111111111111111111111111111111111111111",
+      to:          "0x2222222222222222222222222222222222222222",
+      value:       5000,                    // amount of tokens
+      validAfter:  0,                       // immediately valid
+      validBefore: 1992689033,              // expires at this UNIX timestamp
+      nonce:       "0xabcdefabcdefabcdefabcdefabcdefabcdefabcdefabcdefabcdefabcdefabcd"
+    },
+  };
+
+  let signature = await connectedSigner.signTypedData(
+    approveAgentPayload.domain,
+    approveAgentPayload.types,
+    approveAgentPayload.message,
+  );
+
+  let recoveredAddress = ethers.verifyTypedData(
+    approveAgentPayload.domain,
+    approveAgentPayload.types,
+    approveAgentPayload.message,
+    signature,
+  );
+
+  assertEqual(recoveredAddress, address);
+
+  print(
+    "Turnkey-powered signature - Hyperliquid Sign Transaction Payload:",
+    `${signature}`,
+  );
+}
+
+main().catch((error) => {
+  console.error(error);
+  process.exit(1);
+});
+
+function print(header: string, body: string): void {
```

Comment:
non-blocking: we could probably export this from a single file and use it in multiple examples